CONTRIBUTING.md

```diff
@@ -51,5 +49,45 @@ contributing to. Start with _Step 3_ from above, commenting on the issue to let
 others know you are working on it. If you have any questions related to the
 implementation of the issue, ask them in the issue instead of the PR.
 
+### Branches and Versions
+
+The cuML repository has two main branches: 
+
+1. The Master branch: it contains the last released version and only hot fixes are merged to it.  
+2. Branch-x.y: it is the development branch which contains the upcoming release. All the new features should be based on this branch and Merge/Pull request should target this branch (with the exception of hotfixes).
+    
+### Additional details
+
+There is a new cuML branch called 'branch-0.7', where we will be merging the PRs before the next release. The exceptions to this are the 'hotfixes' which target the issues raised by the Github users and are directly merged to the Master branch. While trying to path an issue which requires a 'hotfix', please state the intent in the PR. A few days before release the current development branch (branch-0.7) will freeze and a new branch, 'branch-0.8' will be created to continue merges.
+
+### Steps for feature development
+
+1. Create a new branch based on `branch-x.y` named following the format `<type>-<ext>-<name>`
+    - Type: 
+        - fea - For if the branch is for a new feature(s)
+        - enh - For if the branch is an enhancement of an existing feature(s)
+        - bug - For if the branch is for fixing a bug(s) or regression(s)
+    - Name: 
+        - A name to convey what is being worked on
+        - Please use dashes or underscores between words as opposed to spaces.
+    
+2. Push whatever development you do to this branch as you would normally and when ready for review create a merge request to the appropriate branch
```

Comment:
Maybe link/point to the "Additional Details" section above, where it outlines which branch is used for the next release.